classes/log/Repository.php

```diff
@@ -107,13 +109,18 @@ public function changeUser(int $oldUserId, int $newUserId)
 
             DB::table('email_log_users')
                 ->where('user_id', $oldUserId)
-                ->whereNotIn('email_log_id', function ($query) use ($newUserId, $oldUserId) {
+                ->whereNotIn('email_log_id', function (Builder $query) use ($oldUserId, $newUserId) {
                     $query->select('t1.email_log_id')
-                        ->from(DB::table('email_log_users')->as('t1'))
-                        ->join(DB::table('email_log_users')->as('t2'), 't1.email_log_id', '=', 't2.email_log_id')
-                        ->where('t1.user_id', $newUserId)
-                        ->where('t2.user_id', $oldUserId);
-                })->update(['user_id' => $newUserId])
+                        ->from(DB::raw('(SELECT email_log_id FROM email_log_users WHERE user_id = ?) as t1'))
+                        ->join(
+                            DB::raw('(SELECT email_log_id FROM email_log_users WHERE user_id = ?) as t2'),
+                            't1.email_log_id',
+                            '=',
+                            't2.email_log_id'
+                        )
+                        ->setBindings([$newUserId, $oldUserId]);
+                })
+                ->update(['user_id' => $newUserId])
```

Comment:
```suggestion
        DB::table('email_log_users AS a')
            ->leftJoin(
                'email_log_users AS b',
                fn (JoinClause $j) => $j->on('b.email_log_id', '=', 'a.email_log_id')->where('b.user_id', '=', $newUserId)
            )
            ->where('a.user_id', $oldUserId)
            ->whereNull('b.user_id')
            ->update(['a.user_id' => $newUserId]);
```

Reply:
Also, as there isn't any kind of "report" over deleted records, I think the `return []` can be dropped. If we ever need a report, it would be better to break this into two methods, instead of guessing that maybe in the future, we'll have one more entry in the array.

Reply:
p.s.: I had to look at the table structure to confirm the purpose of this join, so I think it's useful to add a comment 😊